javascript/ql/src/experimental/Security/CWE-347/JWTMissingSecretOrPublicKeyVerification.help

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+
+<p>The featured CodeQL query warns using of none algorithm in verify() functions imported from jsonwebtoken package developed by the auth0 organization.</p>
+
+<p>Backend JavaScript applications handling JWT could be affected by the none algorithm misconfiguration due to misusing verify() functions imported by jsonwebtoken package.
+Providing an empty string or a false value, instead of a secret or a key, enable the none algorithm to decode JWT payloads without signature verification.
+Misconfigured backend JavaScript on a production environment could be impacted by exploitation violating the integration of a JWT.</p>
```

Comment:
```suggestion
<p>
Applications using JSON Web Token (JWT) can be misconfigured to use the none algorithm.  
The none algorithm is selected by calling the <code>verify()</code> function with a falsy value. 
Using the none algorithm disables the integrity checks of a token, 
and a disabled integrity check might allow an attacker to access or decode information that was supposed to be secret. 
</P>
```

There is no need to mention that this is a CodeQL query.    
And I can also feel that you haven't written that many of these help files before, so I tried rewriting the second paragraph. 
Use it however you want. 

I'm unsure about the last sentence in my rewrite (`might allow an attacker to access or decode information that was supposed to be secret`). Could you elaborate on what an attacker can gain from a misconfigured call to `jwt.verify()`?

Reply:
Thank, changes done.